logstash-core/lib/logstash/runner.rb

```diff
@@ -184,6 +198,21 @@ def configure_plugin_paths(paths)
     end
   end
 
+  def agent_class
+    return @agent_class if @agent_class # We don't want to load these things twice
```

Comment:
I'd personally rather keep the exception out of `LogStash::AgentPluginRegistry.lookup`. I think a nil value better fits a lot of usage patterns.